src/core/pattern.js

```diff
@@ -205,6 +205,8 @@ Shadings.RadialAxial = (function RadialAxialClosure() {
       if (matrix) {
         p0 = Util.applyTransform(p0, matrix);
         p1 = Util.applyTransform(p1, matrix);
+        r0 *= matrix[0];
```

Comment:
Use Util.singularValueDecompose2dScale https://github.com/mozilla/pdf.js/blob/master/src/shared/util.js#L701 instead of the matrix[0]